examples/grayscale_16bits.rs

```diff
@@ -6,12 +6,17 @@ use std::env;
 use std::fs::File;
 use std::io::BufReader;
 
-use image::png::PNGDecoder;
 use image::ImageBuffer;
 use image::Gray16Image;
 use image::imageops::thumbnail;
 
-fn main() {
+
+#[cfg(feature = "png")]
```

Comment:
The proper way to do this is to add the example with a `required-features` entry in Cargo.toml. [More information here](https://doc.rust-lang.org/cargo/reference/manifest.html#the-required-features-field-optional). This required naming the file directly but gives a very neat error message in cargo when trying to execute the example without the feature enabled.

```
[[example]]
name = "grayscale_16bits"
path = "examples/grayscale_16bits.rs"
required-features = ["png"]
```

Reply:
Neat, I'll change that. I had the same issue with unit testing where I had to add some some `#[cfg(feature = "png")]`

Reply:
I just tested. Adding an explicit example configuration to the cargo file will disable example discovery (this is pre-2018 behavior. In 2018 edition, the examples will be auto-discovered).


What I can see:
- Specify all the examples in cargo file
- Keep the feature specific flags in the example source code
- Use 2018 (I supposed this has larger consequences not in scope here...)

Do you have any other suggestion or preference?

```
warning: An explicit [[example]] section is specified in Cargo.toml which currently
disables Cargo from automatically inferring other example targets.
This inference behavior will change in the Rust 2018 edition and the following
files will be included as a example target:

* /home/benoit/PROJECT/image-rs/examples/fractal.rs
* /home/benoit/PROJECT/image-rs/examples/opening.rs
* /home/benoit/PROJECT/image-rs/examples/scaleup/main.rs
* /home/benoit/PROJECT/image-rs/examples/scaledown/main.rs
```

Reply:
I hadn't realized that this was edition specific. Since it is only 4 examples it shouldn't be too complicated to have them entered into `Cargo.toml`.